### ISSUE TITLE
Make Smart Object to Save Logic Connections and Behaviors (SUGGESTION)

### ISSUE THREAD

**Issue by JCLEONITA** [TheGameCreators/GameGuruRepo#4856]
as the title, if we create complex behaviors logic with several objects like the video wehre i made a complete debris system logic for and explòoding barrel , we MUST have samrt objects to save logic linkconnections and behaviors applied to them.. this ways we can create instanceable prefabs of ous creations without needs to redo everything

https://streamable.com/j50gnw

**Comment by MonkeyFrogStudio** [comment]
I agree with this. Basically, whenever we create a "smart object" and save that out to be used in other levels, it needs to come into those levels just as it was saved. As JCLEONITA pointed out, logic connections are not saved with the group. 

As a simple example, I should be able to create a simple switch that opens a door (connected via visual logic nodes), group the door, door frame, and switch, save that group, and use that over and over again in any level. However, the logic nodes connections all get lost when the group is saved. While it's simple enough to wire a switch to a door, there can be very complex logic connections that would be a nightmare to have to rewire every time a group was added to a new level.

**Comment by Necrym59** [comment]
I agree, this is basically creating a prefab object which would help immensly in game level designs
